hedera-node/hedera-app/src/main/java/com/hedera/node/app/SessionContext.java

```diff
@@ -15,5 +15,45 @@
  */
 package com.hedera.node.app;
 
-// FUTURE: Put the per-thread stuff here like parsers
-public record SessionContext() {}
+import static java.util.Objects.requireNonNull;
+
+import com.google.protobuf.Parser;
+import com.hederahashgraph.api.proto.java.Query;
+import com.hederahashgraph.api.proto.java.SignedTransaction;
+import com.hederahashgraph.api.proto.java.Transaction;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * This record keeps a list of everything that is used per-thread
+ *
+ * @param queryParser a parser for {@link Query}
+ * @param txParser a parser for {@link Transaction}
+ * @param signedParser a parser for {@link SignedTransaction}
+ * @param txBodyParser a parser for {@link TransactionBody}
+ */
+public record SessionContext(
+        @NonNull Parser<Query> queryParser,
+        @NonNull Parser<Transaction> txParser,
+        @NonNull Parser<SignedTransaction> signedParser,
+        @NonNull Parser<TransactionBody> txBodyParser) {
+
+    /**
+     * Constructor of {@code SessionContext}
+     *
+     * @param queryParser the {@link Query}-parser
+     * @param txParser the {@link Transaction}-parser
+     * @param signedParser the {@link SignedTransaction}-parser
+     * @param txBodyParser the {@link TransactionBody}-parser
+     */
+    public SessionContext(
```

Comment:
you can use the compact constructor: https://mikemybytes.com/2022/02/16/java-records-and-compact-constructors/

Reply:
Done